deps/rabbit_common/src/supervisor2.erl

```diff
@@ -878,17 +878,13 @@ do_restart_delay(Reason,
 
 maybe_restart(Strategy, Child, State) ->
     case restart(Strategy, Child, State) of
-        {{try_again, Reason}, NState2} ->
+        {{try_again, TryAgainId}, NState2} ->
             %% Leaving control back to gen_server before
             %% trying again. This way other incoming requests
             %% for the supervisor can be handled - e.g. a
             %% shutdown request for the supervisor or the
             %% child.
-            Id = if ?is_simple(State) -> Child#child.pid;
-                    true -> Child#child.id
-                 end,
-            Args = [self(), Id, Reason],
-            {ok, _TRef} = timer:apply_after(0, ?MODULE, try_again_restart, Args),
```

Comment:
`timer:apply_after` with zero timeout executes the function immediatly but in a spawned process. However `try_again_restart/1` only sends a gen_sercer cast so it should be fine to call it from the same process (same way as from `restart/2`)